src/components/button/index.tsx

```diff
@@ -4,37 +4,72 @@ import {
   ButtonProps as ChakraButtonProps,
 } from '@chakra-ui/react';
 
-type WithOnClick<T extends 'submit' | 'button'> = T extends 'submit'
-  ? Pick<ChakraButtonProps, 'onClick'>
-  : { onClick: Exclude<ChakraButtonProps['onClick'], undefined> };
-
 type SharedProps = {
   variant?: 'primary' | 'secondary';
   size?: 'md' | 'lg';
-} & Pick<ChakraButtonProps, 'isDisabled' | 'children'>;
+} & Pick<ChakraButtonProps, 'children'>;
+
+type MandatoryOnClick = {
```

Comment:
I am a bit confused why this type is called `MandatoryOnClick` when `onClick` is optional ? :D

Reply:
Yes, it was a change on the las iteration. The type submit had the property onClick as mandatory but there is the case  on the form where we don't need to pass that property due to the formik.
I could rename them to typeSubmit and typeButton. What do you think?